### PR TITLE
Fixes f4 disco

### PR DESCRIPTION
**Pull request by trns1997**
## Summary
- Add autoleds to STM32F411E-DISCO
- Add the 4 userleds to STM32F411E-DISCO

### CODE REVIEW

File: boards/arm/stm32/stm32f411e-disco/include/board.h [hunk 1]
```diff
@@ -297,19 +297,38 @@
...
 
 /* LED bits for use with board_userled_all() */
 
-#define BOARD_LD2_BIT     (1 << BOARD_LD2)
+#define BOARD_LD3_BIT     (1 << BOARD_LD3)
+#define BOARD_LD4_BIT     (1 << BOARD_LD4)
+#define BOARD_LD5_BIT     (1 << BOARD_LD5)
+#define BOARD_LD6_BIT     (1 << BOARD_LD6)
+
+/* The board has only one controllable LED */
```
Comment (Siviuze) [inline boards/arm/stm32/stm32f411e-disco/include/board.h:322]: comment out of sync? Only LED1 is in use

File: boards/arm/stm32/stm32f411e-disco/src/stm32_autoleds.c [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+  switch (led)
+    {
+    case LED_STARTED:
+    case LED_HEAPALLOCATE:
+      /* As the board provides only one soft controllable LED, we simply
+       * turn it on when the board boots.
+       */
+
+      set_led(true);
```
Comment (Siviuze) [inline boards/arm/stm32/stm32f411e-disco/src/stm32_autoleds.c:81]: No use of others leds?

**Comment by acassis** [comment]
@xiaoxiang781216 there are some issues in the MacOS building:

```
Ignored files:
	include/netutils/cJSON.h
	include/netutils/cJSON_Utils.h
	netutils/cjson/cJSON/
	netutils/cjson/v1.7.10.tar.gz

nothing to commit, working tree clean
  Configuring...
make: *** [olddefconfig] Error 126
  Building NuttX...
clang: error: unknown argument '-mlongcalls'; did you mean '-mlong-calls'?
make[1]: *** No rule to make target `context'.
make: *** [libs/libm/.context] Error 2
/Users/runner/work/nuttx/nuttx/sources/apps/Application.mk:99: "program names  sh does not match mainsrcs nsh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o sh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o  "
/Users/runner/work/nuttx/nuttx/sources/apps/Application.mk:99: "program names  sh does not match mainsrcs nsh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o sh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o  "
make: Target `all' not remade because of errors.
/Users/runner/work/nuttx/nuttx/sources/nuttx/tools/testbuild.sh: line 370: /Users/runner/work/nuttx/nuttx/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  Normalize esp32-devkitc/wifi
make: *** [olddefconfig] Error 126
make: *** [savedefconfig] Error 126
make: *** [olddefconfig] Error 126
HEAD detached at pull/10522/merge
Untracked files:
	kwarning

nothing added to commit but untracked files present
====================================================================================
Configuration/Tool: qemu-intel64/nsh
------------------------------------------------------------------------------------
  Cleaning...
clang: error: unknown argument '-mlongcalls'; did you mean '-mlong-calls'?
clang: error: unknown argument '-mlongcalls'; did you mean '-mlong-calls'?
HEAD detached at pull/10522/merge
Untracked files:
```

and: 


```
In file included from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/sys/types.h:32,
                 from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/stdio.h:30,
                 from ./hello_main.c:26:
Error: /Users/runner/work/nuttx/nuttx/sources/nuttx/include/nuttx/wdog.h:51:20: error: missing binary operator before token "("
   51 | #if UINTPTR_MAX >= UINT32_MAX
      |                    ^~~~~~~~~~
ERROR: x86_64-elf-gcc failed: 1
       command: x86_64-elf-gcc -MT ./hello_main.c.Users.runner.work.nuttx.nuttx.sources.apps.examples.hello.o  -M '-Wall' '-Wstrict-prototypes' '-Wshadow' '-Wundef' '-g' '-O2' '-fno-optimize-sibling-calls' '-fno-omit-frame-pointer' '-fno-crossjumping' '-fno-delete-null-pointer-checks' '-fPIC' '-fno-stack-protector' '-mno-red-zone' '-mrdrnd' '-isystem' '/Users/runner/work/nuttx/nuttx/sources/nuttx/include' '-D__NuttX__' '-D__KERNEL__' '-Wno-cpp' '-Werror' '-pipe' '-I' '/Users/runner/work/nuttx/nuttx/sources/apps/include' ./hello_main.c
/Users/runner/work/nuttx/nuttx/sources/apps/Application.mk:99: "program names  sh does not match mainsrcs nsh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o sh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o  "
In file included from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/sys/types.h:32,
                 from ./string/lib_memcpy.c:26:
Error: /Users/runner/work/nuttx/nuttx/sources/nuttx/include/nuttx/wdog.h:51:20: error: missing binary operator before token "("
   51 | #if UINTPTR_MAX >= UINT32_MAX
      |                    ^~~~~~~~~~
ERROR: x86_64-elf-gcc failed: 1
       command: x86_64-elf-gcc -MT bin/lib_memcpy.o  -M '-Wall' '-Wstrict-prototypes' '-Wshadow' '-Wundef' '-g' '-O2' '-fno-optimize-sibling-calls' '-fno-omit-frame-pointer' '-fno-crossjumping' '-fno-delete-null-pointer-checks' '-fPIC' '-fno-stack-protector' '-mno-red-zone' '-mrdrnd' '-isystem' '/Users/runner/work/nuttx/nuttx/sources/nuttx/include' '-D__NuttX__' '-Wno-cpp' '-Werror' '-pipe' '-I' '/Users/runner/work/nuttx/nuttx/sources/nuttx/libs/libc' ./string/lib_memcpy.c
make[2]: *** [lib_memcpy.ddc] Error 1
In file included from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/sys/types.h:32,
                 from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/sched.h:30,
                 from ./nsh_main.c:28:
Error: /Users/runner/work/nuttx/nuttx/sources/nuttx/include/nuttx/wdog.h:51:20: error: missing binary operator before token "("
   51 | #if UINTPTR_MAX >= UINT32_MAX
      |                    ^~~~~~~~~~
In file included from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/sys/types.h:32,
                 from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/unistd.h:28,
                 from /Users/runner/work/nuttx/nuttx/sources/nuttx/include/getopt.h:31,
                 from ./getopt.c:28:
Error: /Users/runner/work/nuttx/nuttx/sources/nuttx/include/nuttx/wdog.h:51:20: error: missing binary operator before token "("
   51 | #if UINTPTR_MAX >= UINT32_MAX
      |                    ^~~~~~~~~~
```

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 there are some issues in the MacOS building:
> 
> ```
> Ignored files:
> 	include/netutils/cJSON.h
> 	include/netutils/cJSON_Utils.h
> 	netutils/cjson/cJSON/
> 	netutils/cjson/v1.7.10.tar.gz
> 
> nothing to commit, working tree clean
>   Configuring...
> make: *** [olddefconfig] Error 126
>   Building NuttX...
> clang: error: unknown argument '-mlongcalls'; did you mean '-mlong-calls'?
> make[1]: *** No rule to make target `context'.
> make: *** [libs/libm/.context] Error 2
> /Users/runner/work/nuttx/nuttx/sources/apps/Application.mk:99: "program names  sh does not match mainsrcs nsh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o sh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o  "
> /Users/runner/work/nuttx/nuttx/sources/apps/Application.mk:99: "program names  sh does not match mainsrcs nsh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o sh_main.c.Users.runner.work.nuttx.nuttx.sources.apps.system.nsh.o  "
> make: Target `all' not remade because of errors.
> /Users/runner/work/nuttx/nuttx/sources/nuttx/tools/testbuild.sh: line 370: /Users/runner/work/nuttx/nuttx/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
>   Normalize esp32-devkitc/wifi
> make: *** [olddefconfig] Error 126
> make: *** [savedefconfig] Error 126
> make: *** [olddefconfig] Error 126
> HEAD detached at pull/10522/merge
> Untracked files:
> 	kwarning
> 
> nothing added to commit but untracked files present
> 

yes, macOS ci fail recently, but I don't find the patch causes this fail.

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**